exec.go

```diff
@@ -247,15 +247,15 @@ func getProcess(context *cli.Context, bundle string) (*specs.Process, error) {
 	}
 	// Override the user, if passed.
 	if user := context.String("user"); user != "" {
-		u := strings.SplitN(user, ":", 2)
-		if len(u) > 1 {
-			gid, err := strconv.Atoi(u[1])
+		uidString, gidString, doesHaveGid := strings.Cut(user, ":")
+		if doesHaveGid {
+			gid, err := strconv.Atoi(gidString)
 			if err != nil {
 				return nil, fmt.Errorf("bad gid: %w", err)
 			}
 			p.User.GID = uint32(gid)
 		}
-		uid, err := strconv.Atoi(u[0])
+		uid, err := strconv.Atoi(uidString)
 		if err != nil {
 			return nil, fmt.Errorf("bad uid: %w", err)
 		}
```

Comment:
It's common in go to use short variable names when the distance between defining and using them is short; also to [avoid type suffixes](https://google.github.io/styleguide/go/decisions.html#variable-names) (like `String` here`)

In situations like this, `ok` is also a common name (instead of the `doesHaveGid`); while `doesHaveGid` is somewhat more descriptive, it's likely clear from the code what it's handling; using `ok` can reduce cognitive overload, as a reader can recognize the pattern without understanding if `doesHaveGid` has additional purpose or (due to the longer name) possibly used further away from its declaration.

e.g., this could look like;

```go
uid, gid, ok := strings.Cut(user, ":")
if ok {
	v, err := strconv.Atoi(gid)
	if err != nil {
		return nil, fmt.Errorf("bad gid: %w", err)
	}
	p.User.GID = uint32(v)
}
v, err := strconv.Atoi(uid)
if err != nil {
    return nil, fmt.Errorf("bad uid: %w", err)
}
p.User.UID = uint32(v)
```

Somewhat orthogonal to this PR (as it's existing code), but wondering if the `strconv.Atoi(gid)` would be better replaced with `strconv.ParseUint(gid, 10, 32)`, given that we're ultimately casting to a `uint32`; i.e. `strconv.Atoi("4294967296")` will result in gid (uint32) `0`.

Reply:
thanks for your comment. I change the variable names.

Reply:
i did a little benchmark for the second issue you told. and the result is like this:

```
goos: linux
goarch: amd64
pkg: bench_go
cpu: Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz
BenchmarkAtoiToUint32-4   	116347470	        10.19 ns/op
BenchmarkParseUint-4      	62446058	        17.87 ns/op
PASS
ok  	bench_go	3.086s
```
as you can see using strconv.Atoi has better performance than ParseUint.

and also this is the code used for benchmark

```
package main

import (
	"strconv"
	"testing"
)

func AtoiToUint32(s string) uint32 {
	i, _ := strconv.Atoi(s)
	return uint32(i)
}

func ParseUint(s string) uint32 {
	i, _ := strconv.ParseUint(s, 10, 32)
	return uint32(i)
}

func BenchmarkAtoiToUint32(b *testing.B) {
	for i := 0; i < b.N; i++ {
		_ = AtoiToUint32("123456")
	}
}

func BenchmarkParseUint(b *testing.B) {
	for i := 0; i < b.N; i++ {
		_ = ParseUint("123456")
	}
}
```
now with these results, do you think it's better if we change ParseUint to Atoi? and also i didn't know that but [ParseUint return an uint64](https://pkg.go.dev/strconv#ParseUint) so we need uint32 anyway.

Reply:
Thanks for checking that. Hm.. that's a good point. I was hoping it wouldn't make a big difference. It looks like we use the `strconv.ParseUint` approach in some other places already, so it's not unprecedented, and we're still looking at _nanoseconds_ difference, so it's not a _massive_ overhead, but of course every bit stacks up.

I was mostly looking at the "correctness" part here, because invalid values could be silently ignored, which may not be desirable, so perhaps worth the trade-off.

Alternatively, we could manually check for values to be out of range

@kolyshkin any thoughts?


☝️ regardless, I think it's a change we should do in a separate PR (for visibility; to not stuff it into a refactor).